scripts/Course.lua

```diff
@@ -1130,194 +1158,176 @@ end
+        if d and d < distance then
+            return true
+        else
+            return false
+        end
+    end
+    return false
 end
 
 --- Are we closer than distance from the last turn?
 ---@param distance number
 ---@return boolean true when we are closer than distance to the last turn, false otherwise, even
 --- if we can't determine the distance to the last turn.
 function Course:isCloseToLastTurn(distance)
-	return self:isCloseToProperty(distance, Course.getDistanceFromLastTurn)
+    return self:isCloseToProperty(distance, Course.getDistanceFromLastTurn)
 end
 
 --- Are we closer than distance to the next direction change?
 ---@param distance number
 ---@return boolean true when we are closer than distance to the next direction change, false otherwise, or when
 --- the distance is not known
 function Course:isCloseToNextDirectionChange(distance)
-	return self:isCloseToProperty(distance, Course.getDistanceToNextDirectionChange)
+    return self:isCloseToProperty(distance, Course.getDistanceToNextDirectionChange)
 end
 
 function Course:isCloseToLastWaypoint(distance)
-	return self:isCloseToProperty(distance, Course.getDistanceToLastWaypoint)
+    return self:isCloseToProperty(distance, Course.getDistanceToLastWaypoint)
 end
 
 --- Get the length of the up/down row where waypoint ix is located
 --- @param ix number waypoint index in the row
 --- @return number length of the current row
 --- @return number index of the first waypoint of the row or ix, if no turn was found for some reason.
 function Course:getRowLength(ix)
-	for i = ix, 1, -1 do
-		if self:isTurnEndAtIx(i) then
-			return self:getDistanceToNextTurn(i), i
-		end
-	end
-	return 0, ix
+    for i = ix, 1, -1 do
+        if self:isTurnEndAtIx(i) then
+            return self:getDistanceToNextTurn(i), i
+        end
+    end
+    return 0, ix
 end
 
 function Course:getNextRowLength(ix)
-	return self.waypoints[ix].lNextRow
+    return self.waypoints[ix].lNextRow
 end
 
 function Course:getNextRowStartIx(ix)
-	return self.waypoints[ix].nextRowStartIx
+    return self.waypoints[ix].nextRowStartIx
 end
 
 function Course:draw()
-	for i = 1, self:getNumberOfWaypoints() do
-		local x, y, z = self:getWaypointPosition(i)
-		-- TODO_22
-		--cpDebug:drawPoint(x, y + 3, z, 10, 0, 0)
-		Utils.renderTextAtWorldPosition(x, y + 3.2, z, tostring(i), getCorrectTextSize(0.012), 0)
-		if i < self:getNumberOfWaypoints() then
-			local nx, ny, nz = self:getWaypointPosition(i + 1)
-			DebugUtil.drawDebugLine(x, y + 3, z, nx, ny + 3, nz, 0, 0, 100)
-		end
-	end
-end
-
---- Waypoints generated by AD have no explicit reverse attribute, they infer it from elsewhere, see
---- https://github.com/Courseplay/courseplay/issues/7026#issuecomment-808715976
---- We assume all AD courses start forward and assume there's a direction change whenever there's an angle
---- over 100 degrees between two subsequent waypoints
-function Course:addReverseForAutoDriveCourse()
```

Comment:
I need this function later down the road :)

Reply:
Added back in :)